### PR TITLE
Features: MeshGenerator and PropertyController class and its extenstions

### PR DESCRIPTION
**Pull request by LadDeep**
Added MeshGenerator and PropertyController class and its extenstions. Changes should be made based on usage

### CODE REVIEW

File: src/editor/viewport/menu/MeshGenerator.js [hunk 1]
```diff
@@ -8,15 +9,55 @@ export default class MeshGenerator{
...
+            case OBJECT_TYPE.MESH.TORUS:
+                geometry = new THREE.TorusGeometry(10, 5, 10, 30);
+                break; 
+            case OBJECT_TYPE.CAMERA:
+                geometry = new THREE.PerspectiveCamera(75, 1920/1200, 0.1, 1000);
+                break;
+            case OBJECT_TYPE.LIGHT:
+                geometry = new THREE.Light();
+                break; 
+        }
```
Comment (pj-25) [inline src/editor/viewport/menu/MeshGenerator.js:137]: could have change the position to the cursorPoint

File: src/editor/viewport/menu/MeshGenerator.js [hunk 2]
```diff
@@ -8,15 +9,55 @@ export default class MeshGenerator{
...
             CONE: 6,
             TORUS: 7
         },
         CAMERA: 8,
         LIGHT: 9
     };
 
-    static create(objectType){
-        
+    constructor(viewport){
```
Comment (pj-25) [inline src/editor/viewport/menu/MeshGenerator.js]: could have passed initial cursorPoint as argument with default value

**Review by pj-25 — CHANGES_REQUESTED** [review]
minute changes

File: src/editor/viewport/propertyManager/circleProperty.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import PropertyController from "./propertyController";
+
+export default class ConeProperty extends PropertyController{
```
Comment (pj-25) [inline src/editor/viewport/propertyManager/circleProperty.js]: seems u have made mistake in class name

**Review by pj-25 — CHANGES_REQUESTED** [review]
material in propertyController seems not used

File: src/editor/viewport/menu/MeshGenerator.js (hunk 1 above)
Comment (LadDeep) [reply src/editor/viewport/menu/MeshGenerator.js:137]: I've missed that one

**Comment by LadDeep** [comment]
> material in propertyController seems not used



> used for changing color property. Changes are yet to be pushed.

**Comment by pj-25** [comment]
> > material in propertyController seems not used
> 
> > used for changing color property

Ya i do know that, but i think you have not passed material in any of *Property.js

File: src/editor/viewport/propertyManager/boxProperty.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import PropertyController from "./propertyController";
 
 export default class BoxProperty extends PropertyController{
-    constructor(propertyPane, geometry){
+    constructor(propertyPane, geometry, material){
         super(propertyPane);
         this.geometry = geometry;
+        this.material = material;
```
Comment (pj-25) [inline src/editor/viewport/propertyManager/boxProperty.js]: u have used material in PropertyController's constructor, then y don't uh pass it in super()

**Review by pj-25 — CHANGES_REQUESTED** [review]

File: src/editor/viewport/propertyManager/boxProperty.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import PropertyController from "./propertyController";
+
+export default class BoxProperty extends PropertyController{
+    constructor(interactiveMesh){
+        super(interactiveMesh);
+        this.geometryData = this.interactiveMesh.geometry.parameters;
+        this.cubeFolder = this.propertiesPane.addFolder('Add Cube');
+    }
+
+    initCubeProperties(){
```
Comment (pj-25) [inline src/editor/viewport/propertyManager/boxProperty.js:10]: could have used initProperties() in all classes

**Review by pj-25 — COMMENTED** [review]
could have used initProperties() in all classes (common name)

**Pull request merged**